src/Client.php

```diff
@@ -271,26 +271,44 @@ protected function getStream($socket)
     }
 
     /**
-     * Adds an event listener to log data emitted by a stream.
+     * Generates a closure for stream output logging.
      *
-     * @param \Evenement\EventEmitter $emitter
-     * @param \Phergie\Irc\ConnectionInterface $connection Connection
-     *        corresponding to the stream
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @param string $level Evenement logging level to use
+     * @param string $prefix Prefix string for log lines (optional)
+     * @return callable
      */
-    protected function addLogging(EventEmitter $emitter, ConnectionInterface $connection)
+    protected function getOutputLogCallback(ConnectionInterface $connection, $level, $prefix = null)
     {
         $logger = $this->getLogger();
-        $callback = function($msg) use ($logger, $connection) {
-            $mask = sprintf(
-                '%s!%s@%s',
+        if (!method_exists($logger, $level)) {
+            trigger_error("Invalid log level '$level' passed to Client::getOutputLogCallback()", E_USER_ERROR);
+        }
+        $logFunc = [$logger, $level];
+        return function($message) use ($connection, $prefix, $logFunc) {
+            $output = sprintf(
+                '%s!%s@%s %s%s',
                 $connection->getNickname(),
                 $connection->getUsername(),
-                $connection->getServerHostname()
+                $connection->getServerHostname(),
+                $prefix,
+                trim($message)
             );
```

Comment:
Perhaps we should also make this PR dependent on phergie/phergie-irc-connection#6 and modify `getOutputLogCallback()` to use `Connection->getConnectionMask()`?